### DISCUSSION TITLE
why CUDA_VISIBLE_DEVICES="" when preprocess datasets?

### DISCUSSION THREAD

**Discussion by xd2333** [axolotl-ai-cloud/axolotl#1380] [Q&A] [answered]
i found that if i set bf16: true in yml, preprocess will get an error "Value error, bf16 requested, but AMP is not supported on this GPU. Requires Ampere series or above. ". I had to set bf16: auto or remove CUDA_VISIBLE_DEVICES="" to make it run.

so why CUDA_VISIBLE_DEVICES="" when preprocess datasets? will it work correct if i remove it? thanks

**Comment by NanoCode012** [comment, marked as answer]
Hey, you can also run it with single gpu! It's just to prevent multiple gpus trying to preprocess.
